src/Symfony/Bridge/PhpUnit/SymfonyExtension.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\PhpUnit;
+
+use PHPUnit\Runner\Extension\Extension;
+use PHPUnit\Runner\Extension\Facade;
+use PHPUnit\Runner\Extension\ParameterCollection;
+use PHPUnit\TextUI\Configuration\Configuration;
+use Symfony\Bridge\PhpUnit\Extension\DisableClockMockSubscriber;
+use Symfony\Bridge\PhpUnit\Extension\DisableDnsMockSubscriber;
+use Symfony\Bridge\PhpUnit\Extension\EnableClockMockSubscriber;
+use Symfony\Bridge\PhpUnit\Extension\RegisterClockMockSubscriber;
+use Symfony\Bridge\PhpUnit\Extension\RegisterDnsMockSubscriber;
+
+class SymfonyExtension implements Extension
+{
+    public function bootstrap(Configuration $configuration, Facade $facade, ParameterCollection $parameters): void
```

Comment:
this also needs to enable the DebugClassLoader if available, as done in SymfonyTestsListenerTrait.

Reply:
I did not enable the `DebugClassLoader` here on purpose. PHPUnit defers the execution of bootstrapping extensions until the test suite to run was actually loaded. Thus, enabling the class loader here is too late to trigger deprecations for classes that have been loaded before. Right now, the only place to enable it early enough is a custom bootstrap file (like done in #58370).

Reply:
But for projects that don't use a custom bootstrap file, I don't see which custom code can run before the loading of the testsuite. So the autoloading of custom classes will still probably happen _after_ that point.
So the bootstrapping of extensions looks like a good place to register it for a better DX for the simple case (`DebugClassLoader::enable` is already handling the case where autoloaders have already been wrapped by a DebugClassLoader, so it is fine if both the bootstrap file and the extension bootstrapper call it)

Reply:
fair enough, this would still fail if a class (e.g. a fixture) is defined in the same file as a test, but it will probably still cover 99% of the other use cases